CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/blaze/BlazeActions.java

```diff
@@ -113,10 +194,217 @@ public BlazeActions() {
+        colorPanel7 = new JPanel();
+        colorPanel8 = new JPanel();
+        colorPanel9 = new JPanel();
+        colorPanel10 = new JPanel();
+
+        colorPanels = new ArrayList<>();
+        colorPanels.add(colorPanel1);
+        colorPanels.add(colorPanel2);
+        colorPanels.add(colorPanel3);
+        colorPanels.add(colorPanel4);
+        colorPanels.add(colorPanel5);
+        colorPanels.add(colorPanel6);
+        colorPanels.add(colorPanel7);
+        colorPanels.add(colorPanel8);
+        colorPanels.add(colorPanel9);
+        colorPanels.add(colorPanel10);
+
+        final List<ConstellationColor> colors = getPresetCustomColors();
+
+        if (colors != null) {
+            int panelCounter = 0;
+            for (final JPanel panel : colorPanels) {
+                if (panelCounter < colors.size()) {
+                    if (colors.get(panelCounter) != null) {
+                        panel.setBackground(colors.get(panelCounter++).getJavaColor());
+                    }
+                }
+            }
+        }
+
+        final JPanel verticalPanel = new JPanel();
+        verticalPanel.setMaximumSize(new Dimension(200, 75));
+
+        final GridLayout gridLayout = new GridLayout(0, 5, 5, 5);
+        verticalPanel.setLayout(gridLayout);
+        verticalPanel.setBorder(BorderFactory.createTitledBorder("Preset Colors"));
+
+        for (final JPanel colorPanel : colorPanels) {
+            colorPanel.setBackground(DEFAULT_COLOR);
+            verticalPanel.add(colorPanel);
+        }
+
+        colorPanel1.setMaximumSize(new Dimension(5, 5));
+        colorPanel1.addMouseListener(new MouseAdapter() {
+            public void mouseClicked(MouseEvent evt) {
+                colorPanel1MouseClicked(evt);
+            }
+        });
+
+        GroupLayout colorPanel1Layout = new GroupLayout(colorPanel1);
+        colorPanel1.setLayout(colorPanel1Layout);
+        colorPanel1Layout.setHorizontalGroup(colorPanel1Layout.createParallelGroup(GroupLayout.Alignment.LEADING)
+                .addGap(0, 25, Short.MAX_VALUE)
+        );
+        colorPanel1Layout.setVerticalGroup(colorPanel1Layout.createParallelGroup(GroupLayout.Alignment.LEADING)
+                .addGap(0, 25, Short.MAX_VALUE)
+        );
+
+        colorPanel2.setMaximumSize(new Dimension(5, 5));
+        colorPanel2.addMouseListener(new MouseAdapter() {
+            public void mouseClicked(MouseEvent evt) {
+                colorPanel2MouseClicked(evt);
+            }
+        });
+
+        GroupLayout colorPanel2Layout = new GroupLayout(colorPanel2);
+        colorPanel2.setLayout(colorPanel2Layout);
+        colorPanel2Layout.setHorizontalGroup(colorPanel2Layout.createParallelGroup(GroupLayout.Alignment.LEADING)
+                .addGap(0, 23, Short.MAX_VALUE)
+        );
+        colorPanel2Layout.setVerticalGroup(colorPanel2Layout.createParallelGroup(GroupLayout.Alignment.LEADING)
+                .addGap(0, 0, Short.MAX_VALUE)
+        );
```

Comment:
Based on previous comment, potentially some of this could be extracted into a function or two given the creation of each colour panel seems to follow roughly the same pattern

Reply:
Done, good spotting - much neater